projects/export/export/snapshotter.py

```diff
@@ -25,19 +26,31 @@ def add_streaming_input_preprocessor(
     input: "ExposedTensor",
     psd_length: float,
     sample_rate: float,
+    kernel_length: float,
     inference_sampling_rate: float,
     fduration: float,
     fftlength: float,
+    q: Optional[float] = None,
     highpass: Optional[float] = None,
     preproc_instances: Optional[int] = None,
     streams_per_gpu: int = 1,
 ) -> "ExposedTensor":
     """Create a snapshotter model and add it to the repository"""
 
-    batch_size, num_ifos, kernel_size = input.shape
+    batch_size, num_ifos, *kernel_size = input.shape
+    if len(kernel_size) == 2:
```

Comment:
Would a cleaner/more robust safeguard here be something like 

```python
If q is not None:
    if len(kernel_size) != 2:
        raise ValueError("Something is wrong")
    augmentor = ...
```

Reply:
That's a good idea, I'll update